docs/scos/dev/glue-api-guides/202311.0/migrating-old-resource-to-new-infrastructure.md

```diff
@@ -0,0 +1,263 @@
+---
+title: Migration of Legacy Glue API resource to Glue Storefront API infrastructure
+description: The guide will walk you through the process of migration of Legacy Glue API resource to the Glue Storefront API infrastructure
+last_updated: November 29, 2023
+template: glue-api-storefront-guide-template
+---
+## Manual migration of existing Legacy Glue API to the new infrastructure
+
+It’s possible to re-use existing resource code in most of the cases, by adjusting existing code and adding a few more classes from the new infrastructure. Those steps will be define below.
+
+The AvailabilityNotificationRestApi module was used as an example.
+
+For the sake of reducing complexity, the module will be updated instead of creating a new on top. 
+
+### Update dependencies
+
+"spryker/glue-json-api-convention-extension": "^1.0.0" dependency MUST be added, due to the fact that all old resources were bound to the JSON:API convention.
+
+### Create a new controller
+
+Resource controller can only help us with the name (in order to not have names conflict) and with documentation annotations that can be re-used. Action method declaration is not suitable for us, due to different used classes types.
+
+New controller should extend `Spryker\Glue\Kernel\Controller\AbstractStorefrontApiController` class instead of `Spryker\Glue\Kernel\Controller\AbstractController`
+
+Method names can be taken from the original controller due to the same naming patter %http_verb%Action.
+
+### Create a new resource plugin
+
+Create a GlueStorefrontApiApplication directory in the Plugin directory. It will be used to store new resource plugins.
+
+Old resource plugin can be used as a reference.
+
+Abstract class must be changed from `Spryker\Glue\Kernel\AbstractPlugin` to `Spryker\Glue\GlueApplication\Plugin\GlueApplication\AbstractResourcePlugin`
+
+Interface MUST be changed from `Spryker\Glue\GlueApplicationExtension\Dependency\Plugin\ResourceRoutePluginInterface` to `Spryker\Glue\GlueJsonApiConventionExtension\Dependency\Plugin\JsonApiResourceInterface`
```

Comment:
Why is this "MUST" in capitals and the others not?

Reply:
updated